src/controllers/photoController.ts

```diff
@@ -0,0 +1,91 @@
+import { NextFunction, Request, Response } from "express";
+import { deletePhoto, uploadPhoto } from "../services/photoService";
+import User, { IUser } from "../models/User";
+import { NotFoundError, UnauthenticatedError } from "../errors";
+
+export const uploadPhotoController = async (
+  req: Request,
+  res: Response, 
+  next: NextFunction
+): Promise<void> => {
+
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+    if (!req.file) {
+      res.status(400).json({ message: "No file uploaded." });
+      return;
+    }
+
+    const { uploadResult, optimizeUrl, autoCropUrl } = await uploadPhoto(
+      req.file
+    );
+
+    const userProfile = await User.findOne({ "_id": user.userId});
+        
+    if (!userProfile) {
+      return next(new NotFoundError("The user was not found"));
+    }
+
+    userProfile.photoId = uploadResult.public_id;
+    userProfile.photo = optimizeUrl;
+    await userProfile.save();
+
+
+    res.json({
+      message: "Avatar uploaded successfully!",
+      imageUrl: uploadResult.secure_url,
+      optimizedUrl: optimizeUrl,
+      autoCroppedUrl: autoCropUrl,
+    });
+  } catch (error) {
+    console.error(error);
+    res.status(500).json({ message: "Error uploading avatar" });
+  }
+};
+
+export const deletePhotoController = async (
+  req: Request,
+  res: Response,
+  next: NextFunction
+): Promise<void> => {
+  const user: IUser | undefined = req.user;
+
+  if (!user) {
+    return next(new UnauthenticatedError("User is not authenticated"));
+  }
+
+  try {
+    const userProfile = await User.findOne({ "_id": user.userId});
+
+    if (!userProfile) {
+      return next(new NotFoundError("The user was not found"));
+    }
+
+    if (!userProfile.photoId) {
+      return next(new NotFoundError("No photo to delete"));
```

Comment:
NIT: This would probably make more sense as a 400 bad request. a 404 should generally be for requesting a resouce object form the server that doesn't exist (i.e. fetching a user by ID but the ID was not found in the database). Parameters for a query that don't match anything, or a user that doesn't have a profile to delete, are a result of a client sending data that doesn't work and is a bad request.